java-security-it/src/test/java/com/sap/cloud/security/test/integration/XsuaaIntegrationTest.java

```diff
@@ -0,0 +1,78 @@
+package com.sap.cloud.security.test.integration;
+
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import com.sap.cloud.security.config.Service;
+import com.sap.cloud.security.config.cf.VcapServicesParser;
+import com.sap.cloud.security.test.SecurityTestRule;
+import com.sap.cloud.security.token.Token;
+import com.sap.cloud.security.token.TokenClaims;
+import com.sap.cloud.security.token.validation.CombiningValidator;
+import com.sap.cloud.security.token.validation.ValidationResult;
+import com.sap.cloud.security.token.validation.validators.JwtValidatorBuilder;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class XsuaaIntegrationTest {
+
+	@ClassRule
+	public static SecurityTestRule rule = SecurityTestRule.getInstance(Service.XSUAA)
+			.setKeys("/publicKey.txt", "/privateKey.txt");
+
+	@Test
+	public void xsuaaTokenValidationSucceeds_withXsuaaCombiningValidator() throws IOException {
+		OAuth2ServiceConfiguration configuration = VcapServicesParser.fromFile("/xsuaa/vcapServices/singleBinding.json")
+				.createConfiguration();
+		Token token = rule.getPreconfiguredJwtGenerator()
+				.fromFile("/xsuaa/tokens/userAccessTokenRSA256.json")
+				.createToken();
+
+		CombiningValidator<Token> tokenValidator = JwtValidatorBuilder.getInstance(configuration).build();
+		ValidationResult result = tokenValidator.validate(token);
+		assertThat(result.isValid()).isTrue();
+	}
+
+	@Test
+	public void xsaTokenValidationSucceeds_withXsuaaCombiningValidator() throws IOException {
+		OAuth2ServiceConfiguration configuration = VcapServicesParser.fromFile(
+				"/xsuaa/vcapServices/xsaSingleBinding.json")
+				.runInLegacyMode(true)
+				.withUrl(rule.getWireMockServer().baseUrl()) // overridden because of legacy mode
```

Comment:
I do not understand why this is needed? overridden because of legacy mode?

Reply:
In this test the validation runs in legacy mode. In legacy mode the URL for the token_keys endpoint is read from the URL parameter of the configuration instead of the JKU header parameter. That is why the url needs to be overridden with the correct `baseUrl` of the wire mock server.

Reply:
Is the Mockserver setup done in SecurityTestRule? And is there the place, where the urls... (iss, config.url) are replaced?
If yes, we could think about moving the legacyMode flag into that... probably we need then another IT-class...?

Reply:
The Mockserver setup is done in SecurityTestRule. But the SecurityTestRule does not know anything about the service configuration or the VcapServicesParser. Therefore this url needs to be replaced inside the test where all the information is available. The challenge here is that the wireMockServer URL is only available after it has been started.

I also think that it would be easier if we had a component that combines the SecurityTestRule and the configuration object and makes sure that the settings of those two match. Another possibility would be to integrate the configuration and the configuration loading mechanism into the SecurityTestRule. But this would increase the size of the SecurityTestRule further and it is already pretty big.